lib/layer.ts

```diff
@@ -105,3 +105,8 @@ export function getMissingLayerVersionErrorMsg(functionKey: string, formalRuntim
     `Please add the '${paramRuntime}LayerVersion' parameter for the Datadog serverless macro.`
   );
 }
+
+function generateLayerID(lambdaFunctionArn: string, runtime: string) {
+  const layerValue: string = crypto.createHash("sha256").update(lambdaFunctionArn).digest("hex");
+  return layerPrefix + "-" + runtime + "-" + layerValue;
```

Comment:
What is `layerValue` here? It looks like you removed that variable?

Reply:
layerValue used to be a number that would start at 0, be appended to the layerPrefix on the old line 79, and simply be incremented each time a layer was added for each lambda function. This turned out to be a problem because I was getting duplicate layer constructs due to the simple naming convention. To fix this I deleted the layerValue variable in applyLayers and delegated creating the unique Id in the generateLayerId function where the new layerValue was a hash of the lambda functionArn. Ultimately the layer Id was the layer prefix (DatadogLayer) appended to the runtime and the hash value created on line 110.

Reply:
Ah got it, makes sense